sdk/eventhub/event-hubs/src/partitionPump.ts

```diff
@@ -96,10 +92,10 @@ export class PartitionPump {
               return await this.stop(CloseReason.OwnershipLost);
             }
             // this will close the pump and will break us out of the while loop
-            return await this.stop(CloseReason.EventHubException);
+            return await this.stop(CloseReason.Shutdown);
```

Comment:
I might be wrong, but in this case aren't we supposed to have an `OwnershipLost` close reason? The user won't want to checkpoint if an exception has occurred because he might have lost the opportunity to process some events.

Reply:
Its less of whether the user "wants" to checkpoint and more of whether the user "can" checkpoint.

Reply:
In [line - 91](https://github.com/Azure/azure-sdk-for-js/blob/21b18b45a6842e2eba3161c96b94925a9b7c3092/sdk/eventhub/event-hubs/src/partitionPump.ts#L91), I added the check If the exception indicates that the partition was stolen (i.e some other consumer with same ownerlevel started consuming the partition) then close reason should be `OwnershipLost` but if the error is not retryable then we should set the close reason `Shutdown`.

Reply:
Kinda tricky, but I believe I follow your reasoning.